src/vmod_soap_xml.c

```diff
@@ -108,10 +108,20 @@ const char* evaluate_xpath(struct priv_soap_vcl *soap_vcl, struct priv_soap_task
 	for (i = 0; i < (xpathObj->nodesetval ? xpathObj->nodesetval->nodeNr : 0); i++) {
 		if( xpathObj->nodesetval->nodeTab[i]->children &&
 		    xpathObj->nodesetval->nodeTab[i]->children->content ) {
-			return ((const char*)(xpathObj->nodesetval->nodeTab[i]->children->content));
+			// Save free pointer to reset workspace in case of overflow
+			void *f = soap_task->ctx->ws->f;
+			char *res = WS_Copy(soap_task->ctx->ws, xpathObj->nodesetval->nodeTab[i]->children->content, xmlStrlen(xpathObj->nodesetval->nodeTab[i]->children->content) + 1);
```

Comment:
we can replcae xmlStrlen with -1, as WS_Copy do the strlen itself

Reply:
fixed